### PR TITLE
feat(timer): Add a skip for the Timer

### PR DESCRIPTION
**Pull request by bmario**
This makes the metricq::Timer a real deadline timer

### CODE REVIEW

File: src/timer.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+#include "log.hpp"
+
+#include <metricq/logger.hpp>
+#include <metricq/timer.hpp>
+
+#include <fmt/chrono.h>
+
+namespace metricq
+{
+void Timer::timer_callback(std::error_code err)
```
Comment (bmario) [inline src/timer.cpp:69]: Wait, @phijor, you went through the hassle of adding a source file and then only move one method to it?

Comment (phijor) [reply src/timer.cpp:69]: ¯\\\_(ツ)\_/¯

**Review by phijor — APPROVED** [review]
LGTM now, what do you think?

**Review by bmario — COMMENTED** [review]
Github won't allow me to approve this PR, but LGTM 🤷

**Comment by phijor** [comment]
> Github won't allow me to approve this PR, but LGTM shrug

Couldn't request your review either. So yeah.

**Pull request merged**